lib/foreman/http_proxy/net_http_extension.rb

```diff
@@ -2,9 +2,8 @@ module Foreman
   module HttpProxy
     module NetHttpExtension
       def request(req, body = nil, &block)
-        host = URI.parse(@address).host
-        if proxy_http_request?(@proxy_address, host, @socket)
-          log_proxied_request("NetHttp", http_proxy, host)
+        if proxy_http_request?(@proxy_address, address, @socket)
+          log_proxied_request("NetHttp", http_proxy, address)
```

Comment:
if i'm not mistaken, address is an instance variable:
```suggestion
        if proxy_http_request?(@proxy_address, @address, @socket)
          log_proxied_request("NetHttp", http_proxy, @address)
```

Reply:
You are not, but there is also a reader in the Net::HTTP class:

```ruby
      # The DNS host name or IP address to connect to.
      attr_reader :address
```

I thought I would stick with it. No writer tho.